### ISSUE TITLE
Support Docusaurus

### ISSUE THREAD

**Issue by hyuzea** [oven-sh/bun#3426]
### What version of Bun is running?

0.6.10

### What platform is your computer?

ubuntu 22.04

### What steps can reproduce the bug?

 bunx --bun create-docusaurus@latest docusaurus

### What do you see instead?

![image](https://github.com/oven-sh/bun/assets/3823496/dd611005-1e97-40d2-9efc-1a6fe3e6d2e5)

found two bug here, one is the  selection is not interactive, try using up, down key instead  shows  `^[[B^[[A`   
and another else the error in the bottom

**Comment by paperclover** [comment]
this is a combination of #3426 (node:util) and #3205 (raw mode)

**Comment by birkskyum** [comment]
`bunx --bun create-docusaurus@latest docusaurus` is working in bun 0.8.1, but running `start`, `build`, `serve` with `--bun` all returns:

```sh
 9 |    }
10 | 
11 |    let seenVal = false;
12 | 
13 |    // Skip the first stack as it's this function
14 |    stacks.shift();
    ^
TypeError: stacks.shift is not a function. (In 'stacks.shift()', 'stacks.shift' is undefined)
      at /Users/admin/repos/docusaurus-test/docusaurus/node_modules/parent-module/index.js:14:1
      at /Users/admin/repos/docusaurus-test/docusaurus/node_modules/import-fresh/index.js:11:20
      at /Users/admin/repos/docusaurus-test/docusaurus/node_modules/@docusaurus/core/lib/server/config.js:36:10
```

Blocked by
- #4324

**Comment by birkskyum** [comment]
With the stack.shift issue fixed, I now see this:

`bun --bun run start`:

```zig
TypeError: client.terminate is not a function. (In 'client.terminate()', 'client.terminate' is undefined)
      at /Users/admin/repos/docusaurus-test/docusaurus/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:70:12
      at forEach (:1:20)
      at /Users/admin/repos/docusaurus-test/docusaurus/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:64:6
65 |         /**
66 |          * @param {ClientConnection} client
67 |          */
68 |         (client) => {
69 |           if (client.isAlive === false) {
70 |             client.terminate();
                ^
```

`bun --bun run build`:

```zig
➜ bun --bun run build


error: Cannot find module "/Users/admin/repos/docusaurus-test/docusaurus/build" from "bun:main"
```

**Comment by EvHaus** [comment]
I'm seeing the `client.terminate is not a function` error in my custom webpack app as well when running the dev server.

**Comment by birkskyum** [comment]
I tried this with the new docusaurus 3.0.1, and while I still get these server errors, the localhost site does launch properly:

```zig
TypeError: client.ping is not a function. (In 'client.ping(() => {
          })', 'client.ping' is undefined)
      at /Users/admin/repos/bun-kitchensink/docusaurus/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:76:11
      at forEach (:1:21)
      at /Users/admin/repos/bun-kitchensink/docusaurus/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:64:7
65 |         /**
66 |          * @param {ClientConnection} client
67 |          */
68 |         (client) => {
69 |           if (client.isAlive === false) {
70 |             client.terminate();
                 ^
TypeError: client.terminate is not a function. (In 'client.terminate()', 'client.terminate' is undefined)
      at /Users/admin/repos/bun-kitchensink/docusaurus/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:70:13
      at forEach (:1:21)
      at /Users/admin/repos/bun-kitchensink/docusaurus/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:64:7
65 |         /**
66 |          * @param {ClientConnection} client
67 |          */
68 |         (client) => {
69 |           if (client.isAlive === false) {
70 |             client.terminate();
                 ^

```

**Comment by johnnyreilly** [comment]
Does anyone know what the remaining issues are? It seems that some of the original linked issues have been fixed, we're still not in a working state - so is it not known what the remaining issues are?

**Comment by heysokam** [comment]
I found this myself, and _(seems like incorrectly)_ reported the bug in webpack
https://github.com/webpack/webpack/issues/17920

Then researched some more and landed on this issue
It seems to run for me, it just repeatedly spams this error in cli:
```js
✔ Client
  Compiled successfully in 9.78s

client (webpack 5.89.0) compiled successfully
71 |
72 |             return;
73 |           }
74 |
75 |           client.isAlive = false;
76 |           client.ping(() => {});
               ^
TypeError: client.ping is not a function. (In 'client.ping(() => {
          })', 'client.ping' is undefined)
      at /ssd/dev/t/web/docutest/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:76:11
      at forEach (:1:21)
      at /ssd/dev/t/web/docutest/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:64:7
65 |         /**
66 |          * @param {ClientConnection} client
67 |          */
68 |         (client) => {
69 |           if (client.isAlive === false) {
70 |             client.terminate();
                 ^
```

**Comment by heysokam** [comment]
Found something else.
I did `npm create docusaurus` since I figured bun was the one bugging and causing problems.
But then I started docusaurus with `bun start`... and the above errors were not triggered.
So the current problem seems to be something related to `bun create`? :thinking:

**Comment by slorber** [comment]
@Electroid I'm the Docusaurus maintainer and don't know Bun much. 
Can you detail how you run it exactly on a newly initialized site?

To me it looks like `bun run build` or `bun run docusaurus build` will actually use Node.js at runtime? 🤔 

Logging `process.version` will notably log `v20.11.0`, my Node.js version, and I don't see any perf improvement, so I guess this is not the way to test Bun is able to run Docusaurus?


When I try `bun run ./node_modules/.bin/docusaurus build` I get errors

While the following works: `node ./node_modules/.bin/docusaurus build`

![CleanShot 2024-02-05 at 23 07 49@2x](https://github.com/oven-sh/bun/assets/749374/93b6fceb-ff3b-4a61-b019-72c14c43cb05)

(latest Bun v1.0.26)

**Comment by birkskyum** [comment]
> To me it looks like bun run build or bun run docusaurus build will actually use Node.js at runtime? 🤔

You're right. You can force usage of bun with the `--bun` flag, like `bun --bun run ...`

**Comment by slorber** [comment]
Thanks

`bun --run run build` doesn't seem to work

![CleanShot 2024-02-05 at 23 21 00@2x](https://github.com/oven-sh/bun/assets/749374/f6a32347-46ac-4237-8d72-01031d00144c)


`bun --bun run docusaurus build` leads to the same error above:


![CleanShot 2024-02-05 at 23 20 06@2x](https://github.com/oven-sh/bun/assets/749374/16a74bd2-51c2-479b-89ca-7e7bb91f52ea)


---

If that helps, here's a more meaningful error message/stacktrace:

![CleanShot 2024-02-05 at 23 25 26@2x](https://github.com/oven-sh/bun/assets/749374/897853f7-baf7-4c75-9d85-f4253bca4d60)

We are doing SSR/SSG using `vm.Script` through this package: https://github.com/pierrec/node-eval

**Comment by Electroid** [comment]
@slorber Got it, thanks for flagging that. We have some more work to do then

**Comment by slorber** [comment]
Thanks for investigating @Electroid

I'm excited to see Bun support for Docusaurus happening 👍 
We are not the fastest SSG tool out there so having an option to build a site much faster would be super useful to us.

---

Here's a minimal repro of the problem we encounter while rendering static pages of Docusaurus inside `vm.Script`:

https://github.com/slorber/bun-docusaurus-ssg-repro

```js
const eval = require("eval");
const fs = require("fs");

async function run() {
    const fileContent = await fs.promises.readFile("server.bundle.js")
    console.log("fileContent",fileContent)

    const result = eval(
        fileContent,
        /* filename: */ "server.bundle.js",
        /* scope: */ {},
        /* includeGlobals: */ true
    )

    console.log("result",result)
}

run()
```

This runs in Node.js v20 but fails in Bun v1.0.26:

![CleanShot 2024-02-06 at 10 58 23@2x](https://github.com/oven-sh/bun/assets/749374/afa9be49-f33a-42b2-a287-cce1894a79e4)


Of course `server.bundle.js` is a 10k LOC file built by Webpack so it's not "so minimal" 😅 
But at least you do not have to deal with Docusaurus things to troubleshoot the problem.
I hope that helps.

Apart from that issue, it seems that otherwise the Docusaurus init template is able to run under Bun, so it's probably the only remaining problem 👍 

(I still have another issue on our own Docusaurus website, but will have to investigate this independently)

**Comment by slorber** [comment]
Apparently, the issue above is caused by this very old Node.js "require-like" lib:
https://github.com/felixge/node-require-like/blob/master/lib/require-like.js


Used internally in "node-eval", with pseudo-code such as:

```js
const sandbox = {require: requireLike(_filename)}

vmScript.runInNewContext(sandbox,options)
```

https://github.com/pierrec/node-eval/blob/master/eval.js

I thought you'd find this useful, but I'll also try to see on my side if we can get rid of it.

Problematic code in that lib:

```js
var Module = require('module');

Module._load(file, parentModule);
```

Related Bun bug: https://github.com/oven-sh/bun/issues/5925

---

Edit: tried to replace this old `require-like` lib by `Module.createRequire()`. This time I was able to evaluate the script, but I now get another error while trying to build the Docusaurus init site 🤪 

```bash
error: script "docusaurus" was terminated by signal SIGSEGV (Address boundary error)
zsh: segmentation fault  bun --bun run docusaurus build
```

This time I'm not sure how to troubleshoot that 😅 

I also saw these warnings:

```
[bun] Warning: worker_threads.Worker option "eval" is not implemented.
[bun] Warning: worker_threads.Worker option "stdout" is not implemented.
[bun] Warning: worker_threads.Worker option "stderr" is not implemented.
[bun] Warning: worker_threads.Worker option "resourceLimits" is not implemented.
```

**Comment by abrahamcahyoadi** [comment]
I have Problem, Install bun Docusaurus and bun start

Bun version 1.0.30

Error:

```
  Compiled successfully in 27.08s

client (webpack 5.90.3) compiled successfully
Watchpack Error (watcher): Error: AccessDenied
Watchpack Error (watcher): Error: AccessDenied

✔ Client
  Compiled successfully in 669.34ms

client (webpack 5.90.3) compiled successfully
71 |
72 |             return;
73 |           }
74 |
75 |           client.isAlive = false;
76 |           client.ping(() => {});
               ^
TypeError: client.ping is not a function. (In 'client.ping(() => {
          })', 'client.ping' is undefined)
      at /mnt/d/cobalg/docusaurus/docusaurusss/node_modules/@docusaurus/core/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:76:11
      at forEach (:1:21)
      at /mnt/d/cobalg/docusaurus/docusaurusss/node_modules/@docusaurus/core/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:64:7
error: script "start" exited with code 1
```

**Comment by slorber** [comment]
Just tried v1.1 and Docusaurus still can't build.

I get an error due to unimplemented `worker_thread` options (reported here: https://github.com/oven-sh/bun/issues/9816#issuecomment-2032631233) during the CSS/JS minification process.

Disabling modification gets me further until it segfaults during SSG:

```
yarn create docusaurus
bun --bun run docusaurus build --no-minify
```

Looks like it's still not working, similarly to my previous test here: https://github.com/oven-sh/bun/issues/3426#issuecomment-1929173528

**Comment by nlnw** [comment]
It seems to work now for what it's worth. I had no issues with building with `1.1.22`.

**Comment by jimmycallin** [comment]
> Thanks
> 
> `bun --run run build` doesn't seem to work
> 
> ![CleanShot 2024-02-05 at 23 21 00@2x](https://private-user-images.githubusercontent.com/749374/302444547-f6a32347-46ac-4237-8d72-01031d00144c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1NTEzOTUsIm5iZiI6MTcyMzU1MTA5NSwicGF0aCI6Ii83NDkzNzQvMzAyNDQ0NTQ3LWY2YTMyMzQ3LTQ2YWMtNDIzNy04ZDcyLTAxMDMxZDAwMTQ0Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxM1QxMjExMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMzE4MzUwNGQ3OGE1Zjk4MmZiMDExNzU1M2JlMDNjYzJmZGU0M2I2ZDVlYWEwNmE5NzhjZmY5ZjEzMGNkODhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.W1MDjn0DyHB3IbgQ-RU3ce-fi9P4fovkt6lMV6_OQJU)
> 
> `bun --bun run docusaurus build` leads to the same error above:
> 
> ![CleanShot 2024-02-05 at 23 20 06@2x](https://private-user-images.githubusercontent.com/749374/302444385-16a74bd2-51c2-479b-89ca-7e7bb91f52ea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1NTEzOTUsIm5iZiI6MTcyMzU1MTA5NSwicGF0aCI6Ii83NDkzNzQvMzAyNDQ0Mzg1LTE2YTc0YmQyLTUxYzItNDc5Yi04OWNhLTdlN2JiOTFmNTJlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxM1QxMjExMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMGE4ZjAxMjhhMDYzYmE2YTU4ODkxY2M4YmJiZGFlZDIxNWEwYWMyMDQwMGQzODE4OGM3MDkyMzk4NjcwNDIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.60AhAo86LAhDDUlqY_xeRtvkZKhBgcW6WJrmVPS71Po)
> 
> If that helps, here's a more meaningful error message/stacktrace:
> 
> ![CleanShot 2024-02-05 at 23 25 26@2x](https://private-user-images.githubusercontent.com/749374/302445527-897853f7-baf7-4c75-9d85-f4253bca4d60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1NTEzOTUsIm5iZiI6MTcyMzU1MTA5NSwicGF0aCI6Ii83NDkzNzQvMzAyNDQ1NTI3LTg5Nzg1M2Y3LWJhZjctNGM3NS05ZDg1LWY0MjUzYmNhNGQ2MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxM1QxMjExMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMTllNGU0ZWFmZDc5NzcxNzZmOGViOWRiZjNhNjgxMjlkYzIzNjEyZjgwMWQ0NWM1MTNiZjgyYTNmMTkxYzRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.txqb2kB5Hw8m0Lm6QPql8v-rz2xndzQI8FVARv32gro)
> 
> We are doing SSR/SSG using `vm.Script` through this package: https://github.com/pierrec/node-eval

I believe this is the same issue as: https://github.com/oven-sh/bun/issues/11539

**Comment by slorber** [comment]
I also see it working now, but we still need to use `--no-minify` to avoid these errors:

```
[bun] Warning: worker_threads.Worker option "stdout" is not implemented.
[bun] Warning: worker_threads.Worker option "stderr" is not implemented.
[bun] Warning: worker_threads.Worker option "resourceLimits" is not implemented.
```

When using that option, I'm able to build a newly initialized site with `bun --bun run docusaurus build --no-minify`. 


## Benchmark

Surprisingly, it builds slower than Node 22.3:

### Node 22

```bash
hyperfine --runs 5 'yarn clear && yarn build --no-minify'

Benchmark 1: yarn clear && yarn build --no-minify
  Time (mean ± σ):      7.716 s ±  0.080 s    [User: 9.871 s, System: 2.259 s]
  Range (min … max):    7.582 s …  7.793 s    5 runs
```

### Bun 1.1.22

```bash
hyperfine --runs 5 'bun --bun run docusaurus clear && bun --bun run docusaurus build --no-minify'

Benchmark 1: bun --bun run docusaurus clear && bun --bun run docusaurus build --no-minify
  Time (mean ± σ):      8.657 s ±  0.078 s    [User: 12.067 s, System: 1.109 s]
  Range (min … max):    8.584 s …  8.785 s    5 runs
```

**Comment by abrahamcahyoadi** [comment]
bun new version 1.1.26 works.

use powershell

**Comment by o-l-a-v** [comment]
@abrahamcahyoadi 

If anything, Bun v1.1.26 is a step back in my testing. It segfaults when running `bun --bun run docusaurus clear && bun --bun run docusaurus build --no-minify`. Created #13584.

**Comment by slorber** [comment]
Tried again today on v1.1.45

Our Docusaurus v3.7 init template can build using default settings

Still getting those warnings, but it builds:

```bash
NotImplementedError: worker_threads.Worker option "stdout" is not yet implemented in Bun.
NotImplementedError: worker_threads.Worker option "stderr" is not yet implemented in Bun.
NotImplementedError: worker_threads.Worker option "resourceLimits" is not yet implemented in Bun.
```

But it's building slower on Bun than Nodejs 😅 

```bash
hyperfine --prepare 'yarn clear' --runs 3 'yarn build' 'bun --bun run docusaurus build'

Benchmark 1: yarn build
  Time (mean ± σ):     10.881 s ±  0.035 s    [User: 18.228 s, System: 2.983 s]
  Range (min … max):   10.840 s … 10.901 s    3 runs
 
Benchmark 2: bun --bun run docusaurus build
  Time (mean ± σ):     12.743 s ±  0.183 s    [User: 20.936 s, System: 1.526 s]
  Range (min … max):   12.575 s … 12.937 s    3 runs
 
Summary
  yarn build ran
    1.17 ± 0.02 times faster than bun --bun run docusaurus build
```


Couldn't make it work on our docusaurus.io prod website. 
It seems that there are schema config normalization errors (we use Joi ) and some default config values are not applied properly, leading to weird errors downstream

![Image](https://github.com/user-attachments/assets/b182aa27-e43c-4d93-bfcd-21cb84c347c2)

**Comment by Jarred-Sumner** [comment]
@slorber we will investigate a little bit after v1.2 (coming in a few hours). there's still a lot of room for us to improve the performance of `Worker` and `postMessage`

**Comment by slorber** [comment]
Thanks!

If that helps, here are 2 more detailed perf traces comparing Node and Bun.

It seems that the slower part in Bun is the webpack bundling (+2sec) but the rest is a bit faster. Webpack+Terser are probably using workers I guess.


```bash
yarn clear && yarn build
yarn run v1.22.22
$ docusaurus clear
[SUCCESS] Removed the Webpack persistent cache folder at "/Users/sebastienlorber/Desktop/website/node_modules/.cache".
[SUCCESS] Removed the generated folder at "/Users/sebastienlorber/Desktop/website/.docusaurus".
[SUCCESS] Removed the build output folder at "/Users/sebastienlorber/Desktop/website/build".
✨  Done in 1.04s.
yarn run v1.22.22
$ docusaurus build
[PERF] Get locales to build - 63.99 ms - (54mb -> 53mb)
[INFO] [en] Creating an optimized production build...
[PERF] Build > en > Load site > Load plugins > Init plugins - 214.65 ms - (54mb -> 75mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load classic@default - 5.92 ms - (75mb -> 67mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load pages@default > loadContent() - 12.01 ms - (75mb -> 68mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load blog@default > loadContent() - 21.95 ms - (75mb -> 76mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load docs@default > loadContent() - 24.09 ms - (75mb -> 77mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load pages@default > contentLoaded() - 7.07 ms - (69mb -> 77mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load pages@default - 24.36 ms - (75mb -> 77mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load docs@default - 26.96 ms - (75mb -> 77mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load blog@default - 27.70 ms - (75mb -> 77mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content - 28.01 ms - (75mb -> 77mb)
[PERF] Build > en > Load site > Load plugins - 243.59 ms - (54mb -> 77mb)
[PERF] Build > en > Load site - 249.83 ms - (54mb -> 68mb)
[PERF] Build > en > Creating webpack bundler configs - 87.65 ms - (68mb -> 84mb)

✔ Client
  Compiled successfully in 8.76s

✔ Server
  


● Client █████████████████████████ cache (99%) shutdown IdleFileCachePlugin
 stored

✔ Server
  

[PERF] Build > en > Bundling with webpack - 9.71 seconds! - (84mb -> 349mb)
[PERF] Build > en > SSG > Generate static files > Load App renderer > Server bundle size = 1.553 MB
[PERF] Build > en > SSG > Generate static files > Load App renderer > Evaluate server bundle - 52.40 ms - (349mb -> 361mb)
[PERF] Build > en > SSG > Generate static files > Load App renderer - 53.62 ms - (349mb -> 361mb)
[PERF] Build > en > SSG > Generate static files - 273.81 ms - (349mb -> 372mb)
[PERF] Build > en > SSG - 274.40 ms - (349mb -> 372mb)
[WARNING] Will NOT delete server bundle because DOCUSAURUS_KEEP_SERVER_BUNDLE is set to 'true'
[PERF] Build > en > postBuild() - 20.46 ms - (372mb -> 379mb)
[SUCCESS] Generated static files in "build".
[PERF] Build > en - 10.35 seconds! - (53mb -> 379mb)
[PERF] Build - 10.35 seconds! - (53mb -> 379mb)
[INFO] Use `npm run serve` command to test your build locally.
✨  Done in 10.98s.
```


```bash
yarn clear && bun --bun run docusaurus build
yarn run v1.22.22
$ docusaurus clear
[SUCCESS] Removed the Webpack persistent cache folder at "/Users/sebastienlorber/Desktop/website/node_modules/.cache".
[SUCCESS] Removed the build output folder at "/Users/sebastienlorber/Desktop/website/build".
[SUCCESS] Removed the generated folder at "/Users/sebastienlorber/Desktop/website/.docusaurus".
✨  Done in 0.56s.
$ docusaurus build
[PERF] Get locales to build - 66.22 ms - (26mb -> 35mb)
[INFO] [en] Creating an optimized production build...
[PERF] Build > en > Load site > Load plugins > Init plugins - 121.98 ms - (35mb -> 43mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load classic@default > getDefaultCodeTranslationMessages() - 5.64 ms - (43mb -> 43mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load classic@default - 6.69 ms - (43mb -> 43mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load pages@default > loadContent() - 11.28 ms - (43mb -> 43mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load blog@default > loadContent() - 18.61 ms - (43mb -> 43mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load pages@default - 22.20 ms - (43mb -> 45mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load docs@default > loadContent() - 26.24 ms - (43mb -> 45mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load blog@default > contentLoaded() - 5.64 ms - (45mb -> 45mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load blog@default - 28.37 ms - (43mb -> 45mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load docs@default - 29.08 ms - (43mb -> 45mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content - 29.17 ms - (43mb -> 45mb)
[PERF] Build > en > Load site > Load plugins - 152.14 ms - (35mb -> 45mb)
[PERF] Build > en > Load site - 158.12 ms - (35mb -> 45mb)
[PERF] Build > en > Creating webpack bundler configs - 73.24 ms - (45mb -> 52mb)

● Client █████████████████████████ sealing (88%)  
 runtime requirements

✔ Server
  

NotImplementedError: worker_threads.Worker option "stdout" is not yet implemented in Bun.
 code: "ERR_NOT_IMPLEMENTED"

      at warnNotImplementedOnce (internal:shared:26:37)
      at new Worker (node:worker_threads:134:31)
      at initialize (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/workers/NodeThreadsWorker.js:149:20)
      at new ExperimentalWorker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/workers/NodeThreadsWorker.js:145:10)
      at createWorker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/WorkerPool.js:44:12)
      at new BaseWorkerPool (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/base/BaseWorkerPool.js:127:27)
      at new WorkerPool (1:23)
      at new Worker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/index.js:167:26)
      at getWorker (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:362:9)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:450:41)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:390:39)
      at next (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/utils.js:63:7)
      at forEach (1:11)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/utils.js:65:26)
      at new Promise (1:11)
      at throttleAll (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/utils.js:41:10)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:579:11)

NotImplementedError: worker_threads.Worker option "stderr" is not yet implemented in Bun.
 code: "ERR_NOT_IMPLEMENTED"

      at warnNotImplementedOnce (internal:shared:26:37)
      at new Worker (node:worker_threads:134:31)
      at initialize (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/workers/NodeThreadsWorker.js:149:20)
      at new ExperimentalWorker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/workers/NodeThreadsWorker.js:145:10)
      at createWorker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/WorkerPool.js:44:12)
      at new BaseWorkerPool (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/base/BaseWorkerPool.js:127:27)
      at new WorkerPool (1:23)
      at new Worker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/index.js:167:26)
      at getWorker (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:362:9)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:450:41)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:390:39)
      at next (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/utils.js:63:7)
      at forEach (1:11)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/utils.js:65:26)
      at new Promise (1:11)
      at throttleAll (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/utils.js:41:10)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:579:11)

NotImplementedError: worker_threads.Worker option "resourceLimits" is not yet implemented in Bun.
 code: "ERR_NOT_IMPLEMENTED"

      at warnNotImplementedOnce (internal:shared:26:37)
      at new Worker (node:worker_threads:134:31)
      at initialize (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/workers/NodeThreadsWorker.js:149:20)
      at new ExperimentalWorker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/workers/NodeThreadsWorker.js:145:10)
      at createWorker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/WorkerPool.js:44:12)
      at new BaseWorkerPool (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/base/BaseWorkerPool.js:127:27)
      at new WorkerPool (1:23)
      at new Worker (/Users/sebastienlorber/Desktop/website/node_modules/jest-worker/build/index.js:167:26)
      at getWorker (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:362:9)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:450:41)
      at <anonymous> (/Users/sebastienlorber/Desktop/website/node_modules/terser-webpack-plugin/dist/index.js:390:39)

✔ Client
  Compiled successfully in 11.29s

✔ Server
  


● Client █████████████████████████ cache (99%) shutdown IdleFileCachePlugin
 stored

✔ Server
  

[PERF] Build > en > Bundling with webpack - 11.54 seconds! - (52mb -> 313mb)
[PERF] Build > en > SSG > Generate static files > Load App renderer > Server bundle size = 1.553 MB
[PERF] Build > en > SSG > Generate static files > Load App renderer > Evaluate server bundle - 37.88 ms - (313mb -> 313mb)
[PERF] Build > en > SSG > Generate static files > Load App renderer - 38.69 ms - (313mb -> 313mb)
[PERF] Build > en > SSG > Generate static files - 248.35 ms - (313mb -> 313mb)
[PERF] Build > en > SSG - 249.21 ms - (313mb -> 313mb)
[WARNING] Will NOT delete server bundle because DOCUSAURUS_KEEP_SERVER_BUNDLE is set to 'true'
[PERF] Build > en > postBuild() - 37.44 ms - (313mb -> 334mb)
[SUCCESS] Generated static files in "build".
[PERF] Build > en - 12.06 seconds! - (35mb -> 334mb)
[PERF] Build - 12.06 seconds! - (35mb -> 334mb)
[INFO] Use `npm run serve` command to test your build locally.
```

---

I tried using [Docusaurus Faster](https://docusaurus.io/blog/releases/3.6#docusaurus-faster) on our init template (replacing older deps by modern Rust tools: Rspack SWC LightningCSS).

This time it segfaults when bundling

```bash
docusaurus build
[PERF] Get locales to build - 109.06 ms - (26mb -> 42mb)
[INFO] [en] Creating an optimized production build...
[PERF] Build > en > Load site > Load plugins > Init plugins - 137.20 ms - (42mb -> 51mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load classic@default > getDefaultCodeTranslationMessages() - 5.96 ms - (51mb -> 51mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load classic@default - 7.07 ms - (51mb -> 51mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load pages@default > loadContent() - 11.76 ms - (51mb -> 51mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load blog@default > loadContent() - 17.99 ms - (51mb -> 51mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load pages@default - 23.07 ms - (51mb -> 53mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load docs@default > loadContent() - 26.45 ms - (51mb -> 53mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load blog@default - 27.41 ms - (51mb -> 53mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content > Load docs@default - 29.22 ms - (51mb -> 53mb)
[PERF] Build > en > Load site > Load plugins > Load plugins content - 29.32 ms - (51mb -> 53mb)
[PERF] Build > en > Load site > Load plugins - 167.46 ms - (42mb -> 53mb)
[PERF] Build > en > Load site - 173.52 ms - (42mb -> 53mb)
[PERF] Build > en > Creating rspack bundler configs - 22.79 ms - (53mb -> 55mb)
============================================================
Bun Canary v1.1.45-canary.35 (703b9962) macOS Silicon
macOS v15.2
Args: "node" "/Users/sebastienlorber/Desktop/website/node_modules/.bin/docusaurus" "build"
Features: Bun.stdin(2) jsc transpiler_cache(29) tsconfig(5) process_dlopen(4) 
Builtins: "bun:jsc" "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:constants" "node:crypto" "node:dns" "node:events" "node:fs" "node:http" "node:https" "node:module" "node:net" "node:os" "node:path" "node:perf_hooks" "node:process" "node:querystring" "node:readline" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:vm" "node:zlib" "node:worker_threads" "ws" "node:v8" "node:http2" 
Elapsed: 861ms | User: 763ms | Sys: 168ms
RSS: 0.28GB | Peak: 0.28GB | Commit: 1.01GB | Faults: 1744

panic(main thread): Segmentation fault at address 0x5
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.45/Mn2703b996gC9//7//B_+7/kJ+7/kJ___+om5Tmq+yQ+qouHm2/4JA2AK

error: script "docusaurus" was terminated by signal SIGTRAP (Trace or breakpoint trap)
zsh: trace trap  bun --bun run docusaurus build
```
